lib/ios-crash-log.js

```diff
@@ -0,0 +1,68 @@
+import path from 'path';
+import _fs from 'fs';
+import _ from 'lodash';
+import glob from 'glob';
+import { getLogger } from 'appium-support';
+import B from 'bluebird';
+
+let fs = {
+  stat: B.promisify(_fs.statSync),
+  readFile: B.promisify(_fs.readFile)
+};
+require('date-utils');
+let logger = getLogger('ios-crash-log');
+
+class IosCrashLog { 
+  constructor() {
+    this.prevLogs = [];
+    this.logsSinceLastRequest = [];
+    this.crashDir = path.resolve(process.env.HOME, 'Library', 'Logs', 'DiagnosticReports');
+  }
+
+  async getCrashes() {
+    try {
+      let crashFiles = await glob(`${this.crashDir} /*.crash`);
+      return crashFiles;
+    } catch (err) {
+      logger.error('There was a problem getting the crash list');
+      throw err;
+    }
+  }
+
+  async filesToJSON(arr) {
+    let logs = [];
+    for (let i = 0; i < arr.length; i++) {
+      let filename = path.resolve(this.crashDir, arr[i]);
+      let stat = await fs.stat(filename);
+      let logObj = {
+        timestamp: stat.ctime,
+        level: 'ALL', 
+        message: fs.readFile(filename, 'utf8')
+      };
+      logs.push(logObj);
+    }
+    return logs;
+  }
+
+  async startCapture() {
+    let fileList = await this.getCrashes();
+    this.prevLogs = fileList;
+  }
+
+  getLogs() {
+    let crashFiles = glob.sync(`${this.crashDir} /*.crash`);
+    let diff = _.difference(crashFiles, this.prevLogs, this.logsSinceLastRequest);
+    this.logsSinceLastRequest = _.union(this.logsSinceLastRequest, diff);
+    return this.filesToJSON(diff);
+  }
+
+  getAllLogs() {
+    let crashFiles = glob.sync(`${this.crashDir} /*.crash`);
+    let logFiles = _.difference(crashFiles, this.prevLogs);
+    return this.filesToJSON(logFiles);
+  }
+}
+
+export function IosCrashLog() {
```

Comment:
this function has the same name as the class, which I don't think will work. also, the function creates a new object but doesn't do anything with it.

personally I think we should just `export default IosCrashLog;` and be done with it. the user can call `new` just fine.